### DISCUSSION TITLE
SmartIR non crea card con piattaforma fan

### DISCUSSION THREAD

**Discussion by JDrillz** [indomus/forum#3262] [HUB: Home Assistant]
Ciao a tutti, ho necessità di integrare un ventilatore di quelli con luce comandabili solo tramite telecomando economici, ma sto avendo qualche problemino con la creazione della card da parte di SmartIR. 
Ho già provveduto ad acquisire i codici di tutti i tasti con Broadlink su HA e di fatto posso utilizzarlo senza problemi lanciando remote.send_command ed anche creare automazioni. 
Tuttavia vorrei avere una card, decisamente più comoda, ma non riesco.
O meglio, seguendo la vostra guida e provando a crearla come media player mi appare, ma se faccio le stesse cose in piattaforma fan no.

Questo è lo yaml:

```
smartir:

fan:
  - platform: smartir
    name: Ventilatore cucina
    unique_id: ventilatore_cucina
    device_code: 2500
    controller_data: remote.broadlink_rm4_pro
```

mentre questo è il JSON, ovviamente chiamato 2500 e posizionato in \config\custom_components\smartir\codes\fan:

```
{
  "manufacturer": "Acorn",
  "supportedModels": [
    "Unknown"
  ],
  "supportedController": "Broadlink",
  "commandsEncoding": "Base64",
  "velocità": [
    "velocità 1",
    "velocità 2",
    "velocità 3",
	"velocità 4"
  ],
  "commands": {
    "tutto off": "JgBQAAABHZESExITEjcSEhISExMSEhITEzYSNxISEjcSNxM2EzYSNxISEhQSEhISEhQSNhMSEhMSNxI3EjYSNxI3EhMSNxI3EgAE4gABHkgRAA0F",
    "default": {
      "velocità 1": "JgBQAAABHZATExISEjcSExITEhMTERMSEjcTNhITEjYSNxI3EjYTNhM3EhISNxITEjcSNxISEhMTEhI3EhMSNxISEhITNhM3EgAE4gABHkgRAA0F",
      "velocità 2": "JgBQAAABHZISEhISEjcSExMSEhMSEhMTEjcSNhITEjYTNhI3EjcTNRMTEjYTEhITEhITNhIUEhISNxITEjcSNhM2EhITNhI3EgAE4wABHUgRAA0F",
      "velocità 3": "JgBQAAABHZISEhITEzYSEhMTEhISExITEjcSNhITEjcTNhI3EjYTNhITEjYSNxITEzYSNxISEhMTNhITEhMSNxISEhMTNhI3EgAE4gABHUkRAA0F",
      "velocità 4": "JgBQAAABHZESEhITEjYTExISEhMTEhITEjcSNxISEjcSNhM2EjgSNxI2EjcSEhIUEhISNxITEhITEhITEjcSNxI2EhQSNhM2EgAE4gABHkgRAA0F"
    }
  }
}
```

Ho già verificato la validità del JSON con il tool e già riavviato HA. Come dicevo, se lo faccio come media player (modificando il file che usate voi nella guida relativa appunto alla piattaforma SmartIR media player) appare correttamente una card media player.

Mi sfugge qualcosa? Ci sono errori? Grazie

**Comment by KC-inDomus** [comment]
> Mi sfugge qualcosa? Ci sono errori? Grazie

devi crearti un'entità `fan.*`. Se non lo fa, verifica i log di Home Assistant all'avvio.

**Comment by JDrillz** [comment]
Alla fine c'era un "velocità" di troppo al posto di "speed". Adesso è comparsa correttamente.
